torch/distributed/device_mesh.py

```diff
@@ -156,20 +108,18 @@ def create_flatten_mesh(
         ) -> "DeviceMesh":
             root_mesh = _mesh_resources.get_root_mesh(device_mesh)
 
-            flatten_dims_in_root = [
-                not_none(root_mesh.mesh_dim_names).index(flatten_mesh_dim_name)
-                for flatten_mesh_dim_name in not_none(device_mesh.mesh_dim_names)
-            ]
-
             if not mesh_dim_name:
                 mesh_dim_name = "_".join(not_none(device_mesh.mesh_dim_names))
 
+            # Flatten a 1D device mesh into its original mesh_dim_name will return itself.
+            if device_mesh.ndim == 1 and mesh_dim_name in not_none(
+                device_mesh.mesh_dim_names
+            ):
+                return device_mesh
```

Comment:
Do you have to check that `mesh_dim_name` is consistent with the name on the device mesh you directly return here? Otherwise this seems to break the invariant that returned mesh has the name passed by `mesh_dim_name`

Reply:
In this case, because ndim is 1, mesh_dim_names is also just one string list, so  `mesh_dim_name` in device_mesh.mesh_dim_names is sufficient.

Reply:
I'm asking something slightly different. I know that it's valid to use the name from mesh_dim_names. But what if the user requests a different name, that is NOT the one in that list? Now that seems like trouble.